app/src/main/kotlin/com/wire/android/ui/settings/devices/DeviceDetailsViewModel.kt

```diff
@@ -115,17 +116,18 @@ class DeviceDetailsViewModel @Inject constructor(
     private fun observeDeviceDetails() {
         viewModelScope.launch {
             observeClientDetails(userId, deviceId).collect { result ->
-                when (result) {
+                state = when (result) {
                     is GetClientDetailsResult.Failure.Generic -> {
                         appLogger.e("Error getting self clients $result")
-                        state = state.copy(error = RemoveDeviceError.InitError)
+                        state.copy(error = RemoveDeviceError.InitError)
                     }
 
                     is GetClientDetailsResult.Success -> {
-                        state = state.copy(
+                        state.copy(
                             device = Device(result.client),
                             isCurrentDevice = result.isCurrentClient,
-                            removeDeviceDialogState = RemoveDeviceDialogState.Hidden
+                            removeDeviceDialogState = RemoveDeviceDialogState.Hidden,
+                            canBeRemoved = !result.isCurrentClient && isSelfClient && result.client.type == ClientType.Permanent,
```

Comment:
Just thinking out loud: in the future, it would be nice to have this in Kalium. So Kalium can expose a `isRemovable` property for clients.